src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala

```diff
@@ -618,4 +617,31 @@ final case class EServicesApiServiceImpl(
       }
     }
   }
+
+  override def deleteEService(
+    eServiceId: String
+  )(implicit contexts: Seq[(String, String)], toEntityMarshallerProblem: ToEntityMarshaller[Problem]): Route = {
+
+    def checkDraftDescriptorAndSize(eService: CatalogProcessEService): Future[Unit] =
+      getDraftDescriptor(eService) match {
+        case Some(draftDescriptor) => catalogProcessService.deleteDraft(eServiceId, draftDescriptor.id.toString)
+        case None                  => isEServiceDescriptorsEmpty(eService)
+      }
+
+    def isEServiceDescriptorsEmpty(eService: CatalogProcessEService): Future[Unit] = {
+      if (eService.descriptors.isEmpty) catalogProcessService.deleteEService(eService.id)
+      else Future.failed(EServiceCannotBeDeleted(eServiceId))
+    }
+
+    val result: Future[Unit] = for {
+      eServiceUUID <- eServiceId.toFutureUUID
+      eService     <- catalogProcessService.getEServiceById(eServiceUUID)
```

Comment:
Since we get the eService here is it necessary to pass its ID instead of the eService itself?

In the latter case we could skip these lines if we pass the eService directly:

- https://github.com/pagopa/interop-be-catalog-process/blob/1.0.x/src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala#L584
- https://github.com/pagopa/interop-be-catalog-process/blob/1.0.x/src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala#L100

Reply:
Passing a whole object that the called service can get by itself is generally not a great idea: if the called service (in this case, the CatalogProcess) hasn't the rights to get that Service from the management, we're in fact leaking it.